ci/steps/run_tests.py

```diff
@@ -0,0 +1,182 @@
+
+
+@contextlib.contextmanager
+def pushd(new_dir):
+    previous_dir = os.getcwd()
+    os.chdir(new_dir)
+    try:
+        yield
+    finally:
+        os.chdir(previous_dir)
+
+
+def _get_test_suite_from_platform(
+    architecture: str,
+    platform: str,
+):
+    if architecture == 'amd64':
+        if platform == 'ali':
+            return 'ali'
+        elif platform == 'aws':
+            return 'aws'
+        elif platform == 'azure':
+            return 'azure'
+        elif platform == 'gcp':
+            return 'gcp'
+        elif platform == 'openstack':
+            return 'openstack'
+        else:
+            return None
+    else:
+        return None
+
+
+def run_tests(
+    architecture: str,
+    cicd_cfg_name: str,
+    gardenlinux_epoch: str,
+    modifiers: str,
+    platform: str,
+    publishing_actions: str,
+    repo_dir: str,
+    suite: str,
+    snapshot_timestamp: str,
+    version: str,
+    committish: str,
+    pytest_cfg: str = None,
+):
+    print(f'run_test with: {architecture=}, {cicd_cfg_name=}, {gardenlinux_epoch=}')
+    print(f'   : {modifiers=}, {platform=}, {publishing_actions=}, {repo_dir=}')
+    print(f'   : {suite=}, {snapshot_timestamp=}, {version=}')
+    publishing_actions = [
+        glci.model.PublishingAction(action.strip()) for action in publishing_actions.split(',')
+    ]
+
+    if not glci.model.PublishingAction.RUN_TESTS in publishing_actions:
+        print('publishing action "run_tests" not specified - skipping tests')
+        return True
+
+    if os.path.exists('/workspace/skip_tests'):
+        print('Tests already uploaded in previous run, skipping test step')
+        sys.exit(0)
+
+    modifiers = tuple(modifiers.split(','))
+
+    params = TestRunParameters(
+        architecture=architecture,
+        cicd_cfg_name=cicd_cfg_name,
+        gardenlinux_epoch=gardenlinux_epoch,
+        modifiers=modifiers,
+        platform=platform,
+        publishing_actions=publishing_actions,
+        repo_dir=repo_dir,
+        suite=suite,
+        snapshot_timestamp=snapshot_timestamp,
+        version=version,
+        committish=committish
+    )
+
+    params_plugin = PyTestParamsPlugin(params)
+    cfg_name = pytest_cfg.strip()
+    test_cfg_path = os.path.join(repo_dir, "ci", "test_cfg.yaml")
+    if not os.path.exists(test_cfg_path):
+        print(f'Configuration file for tests: {test_cfg_path} not found, exiting')
+        sys.exit(1)
+    if not pytest_cfg:
+        print('No tests configured, nothing to do')
+        sys.exit(0)
+    else:
+        final_arg = None
+        with open(test_cfg_path) as file:
+            test_cfgs = yaml.safe_load(file)
+
+        if  not 'test_cfgs' in test_cfgs:
+            print(f'Profile: {cfg_name} not found in file {test_cfg_path}. Stopping')
+            sys.exit(1)
+        if not cfg_name in test_cfgs['test_cfgs']:
+            print(f'Profile: {cfg_name} not found in file {test_cfg_path}. Stopping')
+            sys.exit(1)
```

Comment:
Can be shortened, since the error-msg is unspecific.
```suggestion
        if  not (cfgs:= test_cfgs.get('test_cfgs')) or not cfg_name in cfgs:
            print(f'Profile: {cfg_name} not found in file {test_cfg_path}. Stopping')
            sys.exit(1)
```